grafana-plugin/src/components/TextEllipsisTooltip/TextEllipsisTooltip.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import { Tooltip } from '@grafana/ui';
+import cn from 'classnames/bind';
+
+import styles from 'assets/style/utils.css';
+
+const cx = cn.bind(styles);
+
+interface TextEllipsisTooltipProps {
+  content: string;
+  queryClassName?: string;
+  placement?: string;
+  className?: string;
+  children: JSX.Element | JSX.Element[];
+}
+
+const TextEllipsisTooltip: React.FC<TextEllipsisTooltipProps> = ({
+  queryClassName = 'overflow-child',
+  className,
+  content: textContent,
+  placement,
+  children,
+}) => {
+  const [isEllipsis, setIsEllipsis] = useState(true);
+  const elContentRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    setEllipsis();
+  }, []);
+
+  const elContent = (
+    <div className={cx(className)} ref={elContentRef} onMouseOver={setEllipsis}>
```

Comment:
ok, we'll display tooltip (if element is too long) `onMouseOver`, but what about calling `setEllipsis ` on `onMouseOut` ?

Reply:
Why would we need to call on mouseleave?

Reply:
ok, just looks a bit weird to update the state on `onMouseHover`, in theory mouse hovering shouldn't affect this state...

Reply:
IMO that's a good spot for checking if anything changed. It's like a trigger to show the tooltip, instead we just update the state which will show the tooltip... I don't like setting it on `resize`, maybe there could some other actions triggering truncation which I'm not aware of right now, `onMouseOver` catches every case.